src/test_rig_bluesky/plans.py

```diff
@@ -38,6 +51,44 @@ def spectroscopy(
     yield from bps.prepare(
         spectroscopy_detector, TriggerInfo(livetime=exposure_time), wait=True
     )
+    # TODO: This would be nicer if NDArrayBaseIO had a PortName signal
+    yield from bps.abs_set(spectroscopy_detector.fileio.nd_array_port, "D2.roistat")
```

Comment:
Must: This should block to avoid race conditions

```suggestion
    yield from bps.abs_set(spectroscopy_detector.fileio.nd_array_port, "D2.roistat", wait=True)
```